code/object_detector/algorithms/object_fill.py

```diff
@@ -75,55 +99,77 @@ def _locate_object(self, frame: np.ndarray, image_size: Vector) -> Optional[Vect
             for x in range(0, int(image_size.x) - 3 * step_size, bound):
                 if y + self._last_center.y < image_size.y:
                     current_y = int(y + self._last_center.y)
-                    if all(self._is_red(z, current_y, frame) for z in range(x, x + bound, step_size)):
+                    if all(self._is_red(z, current_y, frame) and Vector(z, current_y) not in self._blacklisted_pixels
+                           for z in range(x, x + bound, step_size)):
                         return Vector(x + step_size, current_y)
                 if self._last_center.y - y > 0:
                     current_y = int(self._last_center.y - y)
-                    if all(self._is_red(z, current_y, frame) for z in range(x, x + bound, step_size)):
+                    if all(self._is_red(z, current_y, frame) and Vector(z, current_y) not in self._blacklisted_pixels
+                           for z in range(x, x + bound, step_size)):
                         return Vector(x + step_size, current_y)
         return None
 
-    def _is_pixel_on_border(self, pixel: Vector, image_size: Vector):
+    def _is_pixel_on_border(self, pixel: Vector, image_size: Vector) -> bool:
         return pixel.x - self.fill_step_size < 0 or \
                image_size.x - self.fill_step_size <= pixel.x + self.fill_step_size + 1 or \
                pixel.y - self.fill_step_size < 0 or \
                image_size.y - self.fill_step_size <= pixel.y + self.fill_step_size + 1
 
     def _get_neighbours(self, pixel: Vector, image_size: Vector) -> Set[Vector]:
-        if self._is_pixel_on_border(pixel, image_size):
+        if pixel in self._blacklisted_pixels or self._is_pixel_on_border(pixel, image_size):
             return set()
         x_dir_offset = Vector(self.fill_step_size, 0)
         y_dir_offset = Vector(0, self.fill_step_size)
-        return {pixel - x_dir_offset, pixel + x_dir_offset, pixel - y_dir_offset, pixel + y_dir_offset}
-
-    def _fill_get_center(self, object_position: Vector, frame: np.ndarray, image_size: Vector) -> Vector:
+        return {
+            pixel - x_dir_offset,
+            pixel + x_dir_offset,
+            pixel - y_dir_offset,
+            pixel + y_dir_offset
+        }
+
+    def _fill_get_center(self, object_position: Vector, frame: np.ndarray, image_size: Vector) -> Optional[Vector]:
         queue: Deque = deque()
         visited = {object_position}
         for neighbour in self._get_neighbours(object_position, image_size):
             queue.append(neighbour)
             visited.add(neighbour)
         sum_outline = object_position
+        self._blacklisted_pixels.add(object_position)
+        sum_redness = _redness(object_position.x, object_position.y, frame)
         sum_elements_in_outline = 1
         while queue:
             element = queue.popleft()
-            if not self._is_red(element.x, element.y, frame):
+            if element in self._blacklisted_pixels:
```

Comment:
Hvis visited ikke er relevant, så kan man lave minus self._blackliste_pixels i stedet for minus visited, og så er det her tjek ikke nødvendigt